test/bin/eshost.js

```diff
@@ -0,0 +1,379 @@
+    });
+  });
+}
+
+let originalHostConfig;
+let nohosts = '{"hosts":{}}';
+try {
+  if (fs.existsSync(Config.defaultConfigPath)) {
+    originalHostConfig = fs.readFileSync(Config.defaultConfigPath, 'utf-8');
+  } else {
+    originalHostConfig = nohosts;
+  }
+} catch (error) {
+  originalHostConfig = nohosts;
+}
+
+function restoreOriginalHostConfig() {
+  fs.writeFileSync(Config.defaultConfigPath, originalHostConfig);
+}
+
+function emptyHostConfig() {
+  fs.writeFileSync(Config.defaultConfigPath, '{"hosts":{}}');
+}
+
+function toLines(result) {
+  return (result || '').split('\n').map(value => value.trim()).filter(Boolean);
+}
+
+function toHostPath(hostpath) {
+  return path.normalize(hostpath) + (isWindows ? '.exe' : '');
+}
+
+before(function() {
+  restoreOriginalHostConfig();
+});
+
+describe('eshost --help', () => {
+  it('displays help contents', () => {
+    return eshost('--help').then(result => {
+      assert.equal(result.stderr, '');
+
+      // TODO: Determine important things to assert about the help text
+      assert.notEqual(result.stdout, '');
+    });
+  });
+});
+
+describe('eshost --list', () => {
+
+  beforeEach(function() {
+    emptyHostConfig();
+  });
+
+  after(function() {
+    restoreOriginalHostConfig();
+  });
+
+  it('displays empty list when there are no hosts configured', () => {
+    return eshost('--list').then(result => {
+      assert.deepEqual(toLines(result.stdout), [
+        `Using config  ${Config.defaultConfigPath}`,
+        '┌──────┬──────┬──────┬──────┬──────┐',
+        '│ name │ type │ path │ args │ tags │',
+        '└──────┴──────┴──────┴──────┴──────┘'
+      ]);
+    });
+  });
+
+  it('displays host table when there is one host configured', () => {
+    fs.writeFileSync(Config.defaultConfigPath, JSON.stringify({
+      hosts: {
+        js: {
+          type: 'jsshell',
+          path: toHostPath('js')
+        }
+      }
+    }));
+
+    return eshost('--list').then(result => {
+      assert.equal(result.stderr, '');
+      /*
+      │ js   │ jsshell │ /path/to/js │      │      │
+      */
+      assert.ok(result.stdout.includes(' js '));
+      assert.ok(result.stdout.includes('jsshell'));
+      assert.ok(result.stdout.includes(toHostPath('js')));
+    });
+  });
+
+  it('displays host table when there is more than one host configured', () => {
+    fs.writeFileSync(Config.defaultConfigPath, JSON.stringify({
+      hosts: {
+        js: {
+          type: 'jsshell',
+          path: toHostPath('js')
+        },
+        ch: {
+          type: 'ch',
+          path: toHostPath('ch')
+        }
+      }
+    }));
+
+    return eshost('--list').then(result => {
+      assert.equal(result.stderr, '');
+      /*
+      │ js   │ jsshell │ /path/to/js │      │      │
+      ├──────┼─────────┼─────────────┼──────┼──────┤
+      │ ch   │ ch      │ /path/to/ch │      │      │
+      */
+      assert.ok(result.stdout.includes(' js '));
+      assert.ok(result.stdout.includes('jsshell'));
```

Comment:
Or better, don't have the spaces around and check using a regex with word boundaries (`\b`)